chapter_10_production_readiness.asciidoc

```diff
@@ -57,6 +60,38 @@ an ORM, all sorts of middleware, the admin site...
 "What else do you want, a pony?" Well, Gunicorn stands for "Green Unicorn",
 which I guess is what you'd want next if you already had a pony...
 
+
+Gunicorn is a WSGI (usually pronounced as whizgee) HTTP server. Let's go
+through what this means:
+
+* An HTTP server handles incoming HTTP requests and sends back responses to the
+  clients.
+
+* WSGI, the Web Server Gateway Interface, is a set of rules defining how an HTTP
+  server (which can be Gunicorn itself, or Nginx, Apache, etc) and a Python
+  program should communicate.
+
+So, Gunicorn is an HTTP server that knows and uses the WSGI to "talk" to Python
+programs. And these Python programs are likewise expected to work in a specific
+way. You can check out a simple example https://gunicorn.org/[on Gunicorn's
+website] under "Installation".
+https://peps.python.org/pep-3333[If you are interested, you can read the entire
+specification here].
+
+What Django's `runserver` does is it launches a simple HTTP server which is
+designed for development: it handles automatic reloading upon saving of the
+source code and displays verbose information about errors and the entire
+callstack, while serving one user. All of which is amazing for debugging.
+
+And, none of which is ideal for serving real users in a production environment
+and for minimising the chance of leaking sensitive data.
+
+We should use an HTTP server that's designed for the job.
+Let's install Gunicorn!
+
+// CSANAD:  I felt like some explanation regarding what Gunicorn and WSGI are,
```

Comment:
very nice!  i might put it into a sidebar, so it doesn't break up the flow too much...